R/model-module-helpers.R

```diff
@@ -178,23 +178,17 @@ add_pmfs <- function(pmfs) {
   if (!is.list(pmfs)) {
     return(pmfs)
   }
-  lpmfs <- purrr::map_dbl(pmfs, length)
-  l <- sum(lpmfs)
-  conv <- rep(0, l)
-  conv[1:lpmfs[1]] <- pmfs[[1]]
-  for (s in 2:d) {
-    # P(Z = z) = sum_over_x(P(X = x) * P(Y = z - x)) # nolint
-    proc <- rep(0, l)
-    for (i in 1:l) {
-      for (j in 1:lpmfs[s]) {
-        if (i >= j) {
-          proc[i] <- proc[i] + pmfs[[s]][j] * conv[i - j + 1]
-        }
+  return(
+    Reduce(x = pmfs, f = function(conv, pmf) {
+      lc <- length(conv)
+      wd <- seq_len(lc) - 1
+      proc <- numeric(lc + length(pmf))
+      for (j in seq_along(pmf)) {
+        proc[j + wd] <- proc[j + wd] + pmf[j] * conv
       }
-    }
-    conv <- proc
-  }
-  return(conv)
+      proc
```

Comment:
As discussed I would rather see `return` used explicitly here.

Reply:
Fixed in https://github.com/epinowcast/epinowcast/pull/187/commits/bdf172ebeff5eb4b3882cdab6f0e4a3d8abe228f; please check if it follows style.

Reply:
Looks good. Thanks.